### PR TITLE
Adds floor/ceil

### PR DESCRIPTION
**Pull request by dc-dc-dc**
Adds floor/ceil ops to tensor and Onnx ops

Onnx ops after:
190 failed, 654 passed, 1790 skipped, 2 warnings in 15.17s 

Tested on:
- METAL
- CUDA
- GPU
- CPU
- TORCH

### CODE REVIEW

File: tinygrad/runtime/ops_cpu.py [hunk 1]
```diff
@@ -26,9 +26,13 @@ def mulacc(a, b, new_shape):
...
   return mulacc
 
+def match_types(x, y):
+  up = x.dtype if dtypes.from_np(x.dtype).priority > dtypes.from_np(y.dtype).priority else y.dtype
+  return x.astype(up), y.astype(up)
+
 numpy_fxn_for_op: Dict[Op, Callable] = {**base_fxn_for_op, **{
   UnaryOps.NOOP: lambda x: np.require(x, requirements='C'), UnaryOps.EXP2: np.exp2, UnaryOps.LOG2: np.log2, UnaryOps.CAST: lambda x,y: x.astype(y.np), UnaryOps.SIN: np.sin,
-  BinaryOps.MAX: np.maximum, BinaryOps.CMPEQ: lambda x,y: (x==y).astype(np.float32),
+  BinaryOps.MAX: np.maximum, BinaryOps.CMPEQ: lambda x,y: (x==y).astype(np.promote_types(x.dtype,y.dtype)), BinaryOps.MUL: lambda x, y: np.multiply(*match_types(x, y)),
```
Comment (geohot) [inline tinygrad/runtime/ops_cpu.py:35]: I understand the CMPEQ change, but why the multiply one?

File: test/test_ops.py [hunk 2]
```diff
@@ -125,6 +125,8 @@ def test_cmp_lt_backwards(self):
     tt2 = Tensor.ones(4, requires_grad=True)
     self.assertRaises(RuntimeError, (tt1 < tt2).sum().backward)
 
+  def test_floor(self): helper_test_op([(45,65)], lambda x: torch.floor(x), lambda x: x.floor(), forward_only=True)
+  def test_ceil(self): helper_test_op([(45,65)], lambda x: torch.ceil(x), lambda x:x.ceil(), forward_only=True)
```
Comment (geohot) [inline test/test_ops.py:129]: Please be consistent with spacing.

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/codegen/llvmir.py                 1               1
tinygrad/runtime/ops_cpu.py                5               1
tinygrad/tensor.py                         4               0
------------------------------------------------------------
total                                     10               2
------------------------------------------------------------
lines added in the tinygrad folder: 8
```

**Comment by geohot** [comment]
Merged, but we really need the linter to check spacing. And I think overall interpreted backend need a better way to deal with types.

**Pull request merged**